Modules/getpath.py

```diff
@@ -783,6 +783,19 @@ def search_up(prefix, *landmarks, test=isfile):
     base_exec_prefix = config.get('base_exec_prefix') or EXEC_PREFIX or base_prefix
 
 
+# ******************************************************************************
+# MISC. RUNTIME WARNINGS
+# ******************************************************************************
+
+# When running Python from the build directory, if libpython is dynamically
+# linked, the wrong library might be loaded.
+if build_prefix and library and not dirname(abspath(library)).startswith(build_prefix):
+    msg = f'The runtime library has been loaded from outside the build directory ({library})!'
+    if os_name == 'posix':
+        msg += ' Consider setting LD_LIBRARY_PATH=. to force it to be loaded from the build directory.'
+    warn(msg)
```

Comment:
Does it make sense to preemptively error out here and force the user to set up LD_LIBRARY_PATH or a similar mechanism?

The other approach (rpath) was rejected for requiring multiple link jobs but would have enforced the correct library. Making it an error to use the wrong library feels a bit closer in spirit to that. And I'm not sure there's a valid use case for people building a worktree python but loading the wrong library, except when messing up -- and the consequences could be strange forms of crashing.

Reply:
I feel like a warning is enough for now. I think erroring out may possibly be a good next step, but I want to see how this impacts the development workflow for other folks before turning it into a hard error.